src/inference/tests/unit/cpu_map_parser/parser_macos.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <common_test_utils/test_common.hpp>
+
+#include "ie_system_conf.h"
+#include "os/cpu_map_info.hpp"
+
+using namespace testing;
+using namespace ov;
+
+namespace {
+
+#ifdef __APPLE__
+
+struct MacOSCpuMapTestCase {
+    int _processors;
+    int _numa_nodes;
+    int _sockets;
+    int _cores;
+    std::vector<std::vector<int>> _proc_type_table;
+    std::vector<std::pair<std::string, uint64_t>> system_info_table;
+};
+
+class MacOSCpuMapParserTests : public ov::test::TestsCommon,
+                               public testing::WithParamInterface<std::tuple<MacOSCpuMapTestCase>> {
+public:
+    void SetUp() override {
+        const auto& test_data = std::get<0>(GetParam());
+
+        int test_processors = 0;
+        int test_numa_nodes = 0;
+        int test_sockets = 0;
+        int test_cores = 0;
+        std::vector<std::vector<int>> test_proc_type_table;
+
+        ov::parse_processor_info_macos(test_data.system_info_table,
+                                       test_processors,
+                                       test_numa_nodes,
+                                       test_sockets,
+                                       test_cores,
+                                       test_proc_type_table);
+
+        ASSERT_EQ(test_data._processors, test_processors);
+        ASSERT_EQ(test_data._numa_nodes, test_numa_nodes);
+        ASSERT_EQ(test_data._sockets, test_sockets);
+        ASSERT_EQ(test_data._cores, test_cores);
+        ASSERT_EQ(test_data._proc_type_table, test_proc_type_table);
+    }
+};
+
+MacOSCpuMapTestCase test_case_arm = {
```

Comment:
As usual the ask is to add comments for each parameter for the first test case. The comments should explain is it input or expected output of the test and also highlights its meaning. Otherwise it is quite hard to understand what is being tested